.github/workflows/go-code-check.yml

```
@@ -0,0 +1,61 @@
+name: Go Code Quality Check
+on:
+  workflow_call:
+    inputs:
+      go-version:
+        required: false
+        type: string
+        default: ''
+
+      go-version-file:
+        required: false
+        type: string
+        default: 'go.mod'
+
+      working-directory:
+        required: false
+        type: string
+        default: '.'
+
+      golangci-lint-flags:
+        required: false
+        type: string
+        default: '--issues-exit-code=0 --enable exportloopref,goconst,gocritic,godox,gofmt,goimports,gosec,nilerr,noctx,revive,unconvert,unparam'
+
+      run-iac-validation:
+        description: Condition for the IaC 
+        type: boolean
+        required: false
+        default: true
+
+
+jobs:
+  iac-validate:
+    if: inputs.run-iac-validation
+    uses: lifemote/workflows/.github/workflows/iac-validate.yml@feature/add-go-support
```

We can release new version for it.
```suggestion
    uses: lifemote/workflows/.github/workflows/iac-validate.yml@v1.3
```